hashicorp/dev-portal
# Remove edit button only for terraform enterprise

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/terraform/enterprise) 🔎
- [Asana task](https://app.asana.com/0/1207947026228781/1208226840548303/f) 🎟️

## 🗒️ What


Hide "Edit this page on GitHub" link for PTFE since (most) people looking at the page will be unable to do so and clicking the link takes them to a 404.

## 🤷 Why



The initial implementation of this ticket was driven by [this support request](https://hashicorp.slack.com/archives/C5FSPUGDS/p1723236157918749) that came in because a customer had pointed out that the "Edit this page on GitHub" link took them to a 404. This PR was in turn a re-hash of #2540 (which turned out to be _too_ broad and hid the link for **ALL** Terraform docs - which turned out not to be correct) after it was reverted by #2562.

## 🛠️ How



The initial implementation checked for `product.slug + currentPathUnderProduct !== 'terraform/enterprise'`  which _works_ but is a little messy. Shout-out to @heatlikeheatwave who instead proposed checking the JSON returned by `fetchDocument` to determine if the product was PTFE(rather than string matching the URL). They also suggested moving this logic(and related logic for `isPublicContentRepo`) out of server.ts and into the loader(which is much cleaner IMO 🎉)

## 🧪 Testing




1. Navigate to [Terraform docs prod](https://developer.hashicorp.com/terraform/cli) and confirm that the "Edit this page on GitHub" link is visible
2. Navigate to [Terraform docs preview](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/terraform/cli) and verify that the "Edit this page on GitHub" link is visible
3. Navigate to [PTFE docs preview](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/terraform/enterprise) and ensure that the "Edit this page on GitHub" link is not visible
3. Navigate to [HCP docs preview](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/hcp/docs/boundary) and ensure that the "Edit this page on GitHub" link is not visible
3. Navigate to [Sentinel docs preview](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/sentinel/docs/commands) and ensure that the "Edit this page on GitHub" link is not visible
6. Navigate to [the docs preview  for another product](https://dev-portal-git-rm-remove-edit-button-only-for-d17d3b-hashicorp.vercel.app/boundary)(Boundary for example) and ensure that the "Edit this page on GitHub" link is visible


In summary:
- ℹ️ Link **is** visible on developer.hashicorp.com (currently)
- ✅ Link **should** be visible on regular Terraform docs
- ✅ Link **should** be visible on other product pages
- ❌ Link **should not** be visible on PTFE (Private Terraform Enterprise)
- ❌ Link **should not** be visible on HCP
- ❌ Link **should not** be visible on Sentinel
